metaflow/package.py

```diff
@@ -181,6 +223,32 @@ def no_mtime(tarinfo):
         blob[4:8] = [0] * 4  # Reset 4 bytes from offset 4 to account for ts
         return blob
 
+    def wait(self, timeout=DEFAULT_PACKAGE_TIMEOUT):
```

Comment:
I would return True/False. Also, quick path on checking on the bool you have about package readiness. Ideally the triple state:
- None: no state so we wait for the thread
- True: done -- no need to wait on anything
- False: some error happened, no need to wait but no package will be available.

We ideally also have to find a way to report that back with wait (this three state)